src/test/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/TableMappingTest.java

```diff
@@ -353,17 +353,17 @@ void validatePhysicalTable() {
                                         .addSi("physicallsi", LSI, new PrimaryKey("physicalgsihk", N),
                                                 1L)
                                         .build())),
-                "physical table physicalTableName's LSI physicallsi's primary key hashkey must be type S, encountered "
+                "physical table physicalTableName's LSI physicallsi's primary-key hash key must be type S, encountered "
                         + "type N");
     }
 
-    private static void assertException(TestFunction test, String expectedMessage) {
+    private static void assertException(TestFunction test, String expectedMessagePrefix) {
         try {
             test.run();
-            throw new RuntimeException("expected exception '" + expectedMessage + "' not encountered");
+            throw new RuntimeException("expected exception '" + expectedMessagePrefix + "' not encountered");
         } catch (IllegalArgumentException | NullPointerException e) {
-            assertTrue(e.getMessage().equals(expectedMessage) || e.getMessage().startsWith(expectedMessage),
-                    "expected=" + expectedMessage + ", actual=" + e.getMessage());
+            assertTrue(e.getMessage().startsWith(expectedMessagePrefix),
```

Comment:
checking the full message *and* the prefix is redundant (i.e., if s1.equals(s2) then s1.startsWith(s2))

Reply:
well, that code made no sense